discovery_engine_core/core/src/engine.rs

```diff
@@ -272,6 +288,16 @@ impl Engine {
         let providers = Providers::new(&config.clone().into()).map_err(Error::ProviderError)?;
 
         // read the configs
+        let feed_config = FeedConfig {
+            max_docs_per_batch: de_config
+                .extract_inner("feed.max_docs_per_batch")
+                .unwrap_or(config.max_docs_per_feed_batch as usize),
+        };
+        let search_config = SearchConfig {
+            max_docs_per_batch: de_config
+                .extract_inner("search.max_docs_per_batch")
+                .unwrap_or(config.max_docs_per_search_batch as usize),
+        };
```

Comment:
:question: Is there any reason why we make it configurable twice?

- once through the configuration
- once through the deConfig

Reply:
it would be somewhat unexpected if this is not configurable via the `de_config`, since this config is supposed to allow to override every all configurations.

i even thought about removing these two config values from the `InitConfig`, but this should be discussed beforehand, and we have to check if blue uses these at all (as far as i can tell they have hardcoded constant values, so we could just make them internal defaults and just keep the `de_config` override).

Reply:
No particularly strong opinion but I'm more to remove it from the `de_config`, none of the values in the `InitConfig` are configurable using `de_config`, moreover we allows to change these values at runtime like market and excluded sources so I'm more to keep consistency with them and the rest of `InitConfig`.